nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -119,55 +112,46 @@ Costmap2DROS::Costmap2DROS(const std::string & name, tf2_ros::Buffer & tf)
 
   // check if we want a rolling window version of the costmap
   bool rolling_window, track_unknown_space, always_send_full_costmap;
-
-  rolling_window = private_parameters_client->get_parameter<bool>("rolling_window", false);
-  track_unknown_space = private_parameters_client->get_parameter<bool>("track_unknown_space", false);
-  always_send_full_costmap = private_parameters_client->get_parameter<bool>(
-      "always_send_full_costmap", false);
+  this->get_parameter_or<bool>("rolling_window", rolling_window, false);
+  this->get_parameter_or<bool>("track_unknown_space", track_unknown_space, false);
+  this->get_parameter_or<bool>("always_send_full_costmap", always_send_full_costmap, false);
 
   layered_costmap_ = new LayeredCostmap(global_frame_, rolling_window, track_unknown_space);
 
-  if (!private_parameters_client->has_parameter("plugins")) {
-    resetOldParameters(private_nh_);
+  // add and initialize layer plugins
+  if (!parameters_client_->has_parameter("plugin_names") ||
+    !parameters_client_->has_parameter("plugin_types") ) {
+  setPluginParams(node_);
   }
 
-  if (private_parameters_client->has_parameter("plugins")) {
-    XmlRpc::XmlRpcValue my_list;
-    my_list = private_parameters_client->get_parameter<XmlRpc::XmlRpcValue>("plugins");
-
-    for (int32_t i = 0; i < my_list.size(); ++i) {
-      std::string pname = static_cast<std::string>(my_list[i]["name"]);
-      std::string type = static_cast<std::string>(my_list[i]["type"]);
-      RCLCPP_INFO(rclcpp::get_logger("nav2_costmap_2d"), "Using plugin \"%s\"", pname.c_str());
-
+  if (parameters_client_->has_parameter("plugin_names") &&
+    parameters_client_->has_parameter("plugin_types")) {
+    auto param = this->get_parameters({"plugin_names", "plugin_types"});
+    for (int32_t i = 0; i < param[0].get_value<std::vector<std::string>>().size(); ++i) {
+      std::string pname = (param[0].get_value<std::vector<std::string>>())[i];
+      std::string type = (param[1].get_value<std::vector<std::string>>())[i];
+      RCLCPP_INFO(this->get_logger(), "Using plugin \"%s\"", pname.c_str());
       std::shared_ptr<Layer> plugin = plugin_loader_.createSharedInstance(type);
       layered_costmap_->addPlugin(plugin);
-      plugin->initialize(layered_costmap_, name + "/" + pname, &tf_);
+      plugin->initialize(layered_costmap_, name + "_" + pname, &tf_, node_);
```

Comment:
Underscore in naming rather than the namespaced actual name? That would make it hard / impossible for the plugin's to use that name in order to determine its own namespaced parameters in the nested yaml files.